### ISSUE TITLE
Can't delete number if there is one left

### ISSUE THREAD

**Issue by nandorojo** [CaioQuirinoMedeiros/react-native-currency-input#18]
If the input only has one number, deleting it should reset to `0`. However, it just doesn't let you delete.

**Comment by CaioQuirinoMedeiros** [comment]
Hi @nandorojo , can you share a reproducible code?

**Comment by nandorojo** [comment]
Sure, one sec

**Comment by nandorojo** [comment]
Ah, just realized it's because `onValueChange` was returning `null`. Turning that into `0` seemingly fixes it.

```tsx
onChangeValue={value => setState(value || 0)}
```

**Comment by nandorojo** [comment]
There is still an occasional flicker in the case of deleting the last digit:

https://user-images.githubusercontent.com/13172299/142689948-677e3ec5-037d-47bd-8ab5-baec5fdfc272.MOV

But it does work.

It's noticeable on iOS.

Repro: https://snack.expo.dev/@nandorojo/laughing-pretzels

```jsx

import React from 'react';
import CurrencyInput from 'react-native-currency-input';

export default function MoneyInput() {
  const [state, setState] = React.useState(0);

  return (
    <CurrencyInput
      value={state}
      onChangeValue={(value) => setState(value || 0)}
      style={{ margin: 50, fontSize: 30 }}
      prefix="$"
      delimiter=","
      precision={0}
      separator="."
    />
  );
}
```

**Comment by nandorojo** [comment]
It's more noticeable if you set text align to center btw

**Comment by CaioQuirinoMedeiros** [comment]
> Ah, just realized it's because `onValueChange` was returning `null`. Turning that into `0` seemingly fixes it.
> 
> ```tsx
> onChangeValue={value => setState(value || 0)}
> ```

Yeah, I know it's a questionable behavior, but otherwise it would be impossible to clear the input. What you did is correct



> There is still an occasional flicker in the case of deleting the last digit:
> 
>  Upload.from.GitHub.for.iOS.MOV 
> But it does work.
> 
> It's noticeable on iOS.
> 


Unfortunately the flickering is inevitable in such controlled input, see https://reactnative.dev/docs/textinput#value
